mmengine/fileio/backends/aws_backend.py

```diff
@@ -0,0 +1,395 @@
+            'hello world'
+        """
+        return self.get(filepath).decode('utf-8')
+
+    def put(self, obj: bytes, filepath: Union[str, Path]) -> None:
+        """Save data to a given ``filepath``.
+
+        Args:
+            obj (bytes): Data to be saved.
+            filepath (str or Path): Path to write data.
+        """
+        bucket, obj_name = self._parse_path(filepath)
+        self._check_bucket(bucket)
+        with io.BytesIO(obj) as buff:
+            # Todo: add progressPercentage
+            self._client.upload_fileobj(buff, bucket, obj_name)
+
+    def put_text(self,
+                 obj: str,
+                 filepath: Union[str, Path],
+                 encoding: str = 'utf-8') -> None:
+        """Save data to a given ``filepath``.
+
+        Args:
+            obj (str): Data to be written.
+            filepath (str or Path): Path to write data.
+            encoding (str): The encoding format used to encode the ``obj``.
+                Default: 'utf-8'.
+        """
+        self.put(bytes(obj, encoding=encoding), filepath)
+
+    def remove(self, filepath: Union[str, Path]) -> None:
+        """Remove a file from aws s3.
+
+        Args:
+            filepath (str or Path): Path to be removed.
+        """
+        bucket, obj_name = self._parse_path(filepath)
+        self._client.delete_object(Bucket=bucket, Key=obj_name)
+
+    def exists(self, filepath: Union[str, Path]) -> bool:
+        """Check whether a file path exists.
+
+        Args:
+            filepath (str or Path): Path to be checked whether exists.
+        Returns:
+            bool: Return ``True`` if ``filepath`` exists, ``False`` otherwise.
+        """
+        bucket, obj_name = self._parse_path(filepath)
+        return self._check_object(bucket, obj_name)
+
+    def isdir(self, filepath: Union[str, Path]) -> bool:
+        """Check whether a file path is a directory.
+
+        Args:
+            filepath (str or Path): Path to be checked whether it is a
+                directory.
+        Returns:
+            bool: Return ``True`` if ``filepath`` points to a directory,
+            ``False`` otherwise.
+        """
+        bucket, obj_name = self._parse_path(filepath)
+        if self._check_bucket(bucket) and (obj_name.endswith('/')
+                                           or obj_name == ''):
+            return True
+        return False
+
+    def isfile(self, filepath: Union[str, Path]) -> bool:
+        """Check whether a file path is a file.
+
+        Args:
+            filepath (str or Path): Path to be checked whether it is a file.
+        Returns:
+            bool: Return ``True`` if ``filepath`` points to a file, ``False``
+                otherwise.
+        """
+        bucket, obj_name = self._parse_path(filepath)
+        if self._check_bucket(
+                bucket) and not obj_name.endswith('/') and obj_name != '':
+            return True
+        return False
+
```

Comment:
`join_path` function is required to make the backend compatible with CheckpointHook to save models to AWS bucket. See [this](https://github.com/open-mmlab/mmengine/blob/main/mmengine/hooks/checkpoint_hook.py#L275). You may simply take the one from PetrelBackend and it will work.